lib/down_time_checker.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class DownTimeChecker
+  DOWNTIME_WINDOWS = [
+    ['Tuesday at 2am UTC', 'Tuesday at 10am UTC'],
+    ['Wednesday at 2am UTC', 'Wednesday at 10am UTC'],
+    ['Thursday at 2am UTC', 'Thursday at 10am UTC'],
+    ['Friday at 2am UTC', 'Friday at 10am UTC'],
+    ['Saturday at 2am UTC', 'Saturday at 10am UTC']
+  ].freeze
+
+  def initialize(service)
+    @service = service
+  end
+
+  def down?
+    if DOWNTIME_WINDOWS.any? { |w| (parse_time(w[0])..parse_time(w[1])).cover?(Time.now.utc) }
```

Comment:
Could you refactor this to be clearer?

Maybe extract this line to:
```suggestion
    if within_downtime_window?
```

```ruby
def within_downtime_window?
 DOWNTIME_WINDOWS.any? do |window|
    window = downtime_window window
    window.cover? Time.now.utc
  end
end

def downtime_window(begin_end_time)
  begin_time = Time.zone.parse begin_end_time.first
  end_time = Time.zone.parse begin_end_time.last

  begin_time..end_time
end
```

Reply:
done